timelock-api/src/main/conjure/timelock-api.yml

```diff
@@ -217,6 +217,40 @@ types:
       LeaderTimes:
         fields:
           leaderTimes: map<Namespace, LeaderTime>
+      TimestampName:
+        alias: string
+        safety: safe
+      NamedMinTimestampLeaseRequest:
+         fields:
+           timestampName: TimestampName
+           requestId:
+             type: uuid
+             safety: safe
+           numFreshTimestamps:
+             type: integer
+             safety: safe
+      NamedMinTimestampLeaseRequests:
+         alias: list<NamedMinTimestampLeaseRequest>
```

Comment:
Ideally, the client only sends one request per timestampName. But, if they don't that's fine, on the server side, we will treat each element separately.

Also, it is possible to use a map here (from timestamp name to a request object with no timestamp name). I had some reason for this earlier today but i forgot - let me know of what you think works best

Reply:
A reason (don't think it was what I had in mind) to use list is perhaps that you are not forced to have one request only per timestampName (I presume having one big request with many fresh ts is more likely to run into the paxos timestamp bound, maybe it's better to partially succeed than to completely fail in those cases)